main.go

```diff
@@ -209,39 +492,104 @@ func runClient(host string) {
 		Transport: tr,
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	r := &netperfReader{doneCh: ctx.Done()}
-	r.buf = make([]byte, 1*humanize.MiByte)
-	rand.Read(r.buf)
+	var index string
+	for {
+		time.Sleep(1 * time.Second)
+		index = getWriterIndex(r, clnt)
+		if index == "" {
+			globalErr = errors.New("Waiting for index ...")
+			if ctx.Err() != nil {
```

Comment:
```suggestion
			if ctx.Err() != nil {
```

we can certainly do

```
select {
   case <-ctx.Done():
}
```

instead of `if ctx.Err() != nil {`

Reply:
With `default:`, yes.  `ctx.Err()` is not defined before `ctx.Done` has fired.

Reply:
It seems ctx.Err() is always set when ctx.Done() is called.. I think it's cleaner just to check if Err() != nil then to write a whole select everytime I want to check the context.